File: hw/aimacana/u2/hw18mongodb/python/model/product.py
from decimal import Decimal
from model.product_category import ProductCategory

class Product:
    def __init__(self, product_id: str = "", name: str = "", price: Decimal = Decimal('0.0'), category: ProductCategory = ProductCategory.HOT_BEVERAGE):
        self.product_id = product_id
        self.name = name
        self.price = price
        self.category = category

    def to_dict(self) -> dict:
        return {
            "productId": self.product_id,
            "name": self.name,
            "price": float(self.price),
            "category": self.category.value
        }

    @classmethod
    def from_dict(cls, data: dict):
        if not data:
            return None
        return cls(
            product_id=data.get("productId", ""),
            name=data.get("name", ""),
            price=Decimal(str(data.get("price", 0.0))),
            category=ProductCategory(data.get("category", "HOT_BEVERAGE"))
        )

    def __str__(self) -> str:
        return f"Product{{productId='{self.product_id}', name='{self.name}', price={self.price}, category={self.category.name}}}"

File: hw/aimacana/u2/hw18mongodb/typescript/src/model/product.ts
import { ProductCategory } from './productCategory.js';

export class Product {
    public productId: string;
    public name: string;
    public price: number;
    public category: ProductCategory;

    constructor(
        productId = '',
        name = '',
        price = 0,
        category: ProductCategory = ProductCategory.HOT_BEVERAGE
    ) {
        this.productId = productId;
        this.name = name;
        this.price = price;
        this.category = category;
    }

    public static fromDocument(doc: any): Product | null {
        if (!doc) return null;
        return new Product(
            doc.productId,
            doc.name,
            doc.price,
            doc.category as ProductCategory
        );
    }
}
